services/app/src/contexts/ChessBoardContext.jsx

```diff
@@ -0,0 +1,104 @@
+/* eslint-disable react/jsx-no-constructed-context-values */
+/* eslint-disable react/prop-types */
+import React, { createContext, useState } from 'react';
+import { useChessBoard } from '../hooks/useChessBoard';
+import { NUMBER_OF_ROWS, NUMBER_OF_COLUMNS } from '../constants/board';
+import '../static/style/gameScore.css';
+
+export const ChessBoardContext = createContext();
+let num;
+
+const ChessBoardContextProvider = ({ children }) => {
+  const [selectedItem, setSelectedItem] = useState('');
+  const [board, setBoard] = useChessBoard();
+
+  const placeItemOnBoard = (row, tile) => {
+    if (selectedItem) {
+      const copyOfBoard = [...board];
+      copyOfBoard[row][tile].items.push(
+        selectedItem,
+        calculateTheScore(row, tile, selectedItem),
+        // + calculateTheScore(row, tile, selectedItem),
+      );
+      setBoard(copyOfBoard);
+      setSelectedItem('');
+    }
+  };
+
+  const calculateTheScore = (row, tile, selectedItem) => {
```

Comment:
I think it would be a good idea to move this function to some other file as it does not depend on any state that is managed by this context.